File: python_open/vecOp.py
import cv2
import numpy as np

def main():
    # Vec3b와 Scalar 예시
    p1 = np.zeros(3, dtype=np.uint8)  # 3채널 BGR 벡터 초기화
    p2 = np.array([0, 0, 255], dtype=np.uint8)  # 빨간색

    p1[0] = 255  # 파란색 채널 설정

    print("p1 ->", p1)
    print("p2 ->", p2)

    gray = np.array([128], dtype=np.uint8)  # 회색
    print("gray ->", gray)

    # Airy Blue로 초기화된 Mat 객체 생성
    img1 = np.full((400, 400, 3), (240, 248, 255), dtype=np.uint8)  # 3채널 (BGR)
    img2 = np.full((400, 400), 200, dtype=np.uint8)                   # 1채널 (Gray)
    img3 = np.full((400, 400, 3), (240, 248, 255), dtype=np.uint8)  # 3채널 (BGR)
    img4 = np.full((400, 400), 200, dtype=np.uint8)                   # 1채널 (Gray)
    img5 = np.full((400, 400, 3), (240, 248, 255), dtype=np.uint8)  # 3채널 (BGR)

    # 각 이미지를 출력
    cv2.imshow("img1", img1)
    cv2.imshow("img2", img2)
    cv2.imshow("img3", img3)
    cv2.imshow("img4", img4)
    cv2.imshow("img5", img5)

    cv2.waitKey(0)  # 키 입력을 기다림
    cv2.destroyAllWindows()  # 모든 창 닫기

if __name__ == "__main__":
    main()

File: cppOpen/vecOp.cpp
#include "opencv2/opencv.hpp"
#include <iostream>

using namespace cv;
using namespace std;

int main() 
{
    Vec3b p1, p2(0, 0, 255);
    p1[0] = 255;

    cout << "p1 -> " << p1 << endl;
    cout << "p2 -> " << p2 << endl;

    Scalar gray = 128;
    cout << "gray -> " << gray << endl;

    // Airy Blue로 초기화된 Mat 객체 생성
    Mat img1 = Mat(400, 400, CV_8UC3, Scalar(240, 248, 255));  // 3채널 (BGR)
    Mat img2 = Mat(400, 400, CV_8UC1, Scalar(200));             // 1채널 (Gray)
    Mat img3 = Mat(400, 400, CV_8UC3, Scalar(240, 248, 255));  // 3채널 (BGR)
    Mat img4 = Mat(400, 400, CV_8UC1, Scalar(200));             // 1채널 (Gray)
    Mat img5 = Mat(400, 400, CV_8UC3, Scalar(240, 248, 255));  // 3채널 (BGR)

    // 각 이미지를 출력
    imshow("img1", img1);
    imshow("img2", img2);
    imshow("img3", img3);
    imshow("img4", img4);
    imshow("img5", img5);

    waitKey(0);  // 키 입력을 기다림
    return 0;
}
